source/views/components/filter/filter-popover.js

```diff
@@ -0,0 +1,65 @@
+//@flow
+import * as React from 'react'
+import {StyleSheet, View} from 'react-native'
+import Popover from 'react-native-popover-view'
+import {FilterSection} from './section'
+import type {FilterType} from './types'
+import * as c from '../colors'
+
+type Props = {
+	anchor: any,
+	filter: FilterType,
+	onClosePopover: (filter: FilterType) => any,
+	visible: boolean,
+}
+
+type State = {
+	filter: FilterType,
+}
+
+export class FilterPopover extends React.PureComponent<Props, State> {
+	state = {
+		filter: this.props.filter,
+	}
+
+	static getDerivedStateFromProps(props: Props) {
+		return {
+			filter: props.filter,
+		}
+	}
+
+	onFilterChanged = (filter: FilterType) => {
+		this.setState(() => ({filter: filter}))
+	}
+
+	render() {
+		const {filter} = this.state
+		const {anchor, onClosePopover, visible} = this.props
+		return (
+			<Popover
+				arrowStyle={arrowStyle}
+				fromView={anchor}
+				isVisible={visible}
+				onClose={() => {
+					onClosePopover(filter)
+				}}
+				placement="bottom"
+			>
+				<View style={styles.popoverContainer}>
+					<FilterSection filter={filter} onChange={this.onFilterChanged} />
```

Comment:
Can we make it so we can pass `style` to FilterSection instead of wrapping it in a view?

Reply:
I suppose...but then we'd have to wrap all three types of `FilterSection`s with another View, or pass the style prop all the way down to them, which seems like a lot. But if that's best practice, I guess we can do that. It's just more code.